Assets/Scripts/Game/UserInterfaceWindows/DaggerfallSpellBookWindow.cs

```diff
@@ -253,32 +255,43 @@ protected virtual void RefreshSpellsList(bool preservePosition)
             }
         }
 
-        protected virtual void PopulateSpellsList(List<EffectBundleSettings> spells, int? availableSpellPoints = null)
+        protected virtual List<int> PopulateSpellsList(List<EffectBundleSettings> spells, int? availableSpellPoints = null)
```

Comment:
I'm not too keen of changing the interface of `protected` access methods. It will break mods who override this function, which is not something DFU is meant to do at this stage.
I can't think of another way to do this without that, except by letting `PopulateSpellsList` set `spellBookIndices` from within, and have all the other methods check `spellBookIndices` for null, in case a mod replaced this function without updating

Reply:
Does such mod exist? Because the implicit invariant that generated spell listbox must use the same indices as the spellbook seriously limited the usefulness of overriding that method.

I did not modify spellbookIndices directly because that method is also used in buymode == true, when populating spellbookIndices would make no sense. But I'm not particularly fond of the solution I found either.

Reply:
Should be better now